### PR TITLE
feat: Add a helper generic function DecodeSliceOfJSON

### LINKED ISSUE #521: Add DecodeSliceOfJSON to handle []Struct results easier
(init idea: Add *RedisMessage.ScanSlice(dst any) to handle []Struct results easier)

When values of key1, key2 are both instances of X, code below returns an error of course because DecodeJSON handles only string

```go
type X struct {
	Name string
}

xs := make([]*X, 0)
if err := redisClient.Do(ctx, s.redisClient.B().Mget().Key("key1", "key2").Build()).DecodeJSON(xs); err != nil {
    return err
}
```

I could not find any method to scan a slice of struct so I would like to add one, like "ScanSlice". 
Scanning slice of struct in golang is not always easy so providing a scanning-slice method would be helpful.

some libraries, for example, sqlx.DB has a method "SelectContext" to scan a slice of struct and this is really helpful for us.

```go
xs := make([]*X, 0)
if err := sqlx.DB.SelectContext(ctx, &xs, q, args...); err != nil { return err }

// -> xs = []&X{{Name: "name1"}, {Name: "name2"}, ....}
```

### PR DESCRIPTION
**Pull request by sgasho**
added a helper function DecodeSliceOfJSON to struct scan a redis result.
code provided in https://github.com/redis/rueidis/issues/521#issuecomment-2029762737 did not work so I changed a little.

## Related Issue
close #521

### CODE REVIEW

File: helper.go [hunk 1]
```diff
@@ -181,6 +181,25 @@ func JsonMSet(client Client, ctx context.Context, kvs map[string]string, path st
...
+		return err
+	}
+
+	ts := make([]T, len(values))
+	for i, v := range values {
+		var t T
+		if err = v.DecodeJSON(&t); err != nil && !IsRedisNil(err) {
+			return err
+		}
+		ts[i] = t
```
Comment (rueian) [inline helper.go:197]:
```suggestion
func DecodeSliceOfJSON[T any](result RedisResult, dest *[]*T) error {
	values, err := result.ToArray()
	if err != nil {
		return err
	}

	ts := make([]*T, len(values))
	for i, v := range values {
		var t T
		if err = v.DecodeJSON(&t); err != nil && !IsRedisNil(err) {
			return err
		}
		ts[i] = &t
```

Considering the case of Nil, I think accepting a `*[]*T` is still a better approach.

**Comment by rueian** [comment]
image
The error handling parts is not tested. Could you also add some tests for them?

**Comment by rueian** [comment]
Sorry @sgasho, I didn't notice that your original version can actually work with both `*[]T` and `*[]*T`. So, I now believe your original version is superior. Could you change it back?

**Review by rueian — APPROVED** [review]

File: helper.go [hunk 2]
```diff
@@ -181,6 +181,28 @@ func JsonMSet(client Client, ctx context.Context, kvs map[string]string, path st
...
+	if err != nil {
+		return err
+	}
+
+	ts := make([]T, len(values))
+	for i, v := range values {
+		var t T
+		if err = v.DecodeJSON(&t); err != nil {
+			if IsRedisNil(err) {
+				continue
```
Comment (ash2k) [inline helper.go:196]: @rueian @sgasho If this happens, then `ts[i]` will be an empty value. Is this the expected behavior? Probably not.

Comment (ash2k) [reply helper.go:196]: Ah, I see there is a test for this behavior, so probably it is a desired behavior. Sorry for the noise.

**Pull request merged**